scripts/zkai_attention_kv_fused_softmax_table_route_matrix_gate.py

```
@@ -0,0 +1,760 @@
+    if len(signature.parameters) == 1:
+        module.validate_result(gate_result)
+        return
+    source_input = module.read_bounded_json(module.SOURCE_INPUT_JSON, module.MAX_SOURCE_INPUT_JSON_BYTES, "source input")
+    envelope = module.read_bounded_json(module.FUSED_ENVELOPE_JSON, module.MAX_FUSED_ENVELOPE_JSON_BYTES, "fused envelope")
+    module.validate_result(gate_result, envelope, source_input)
+
+
+def source_dimensions(source_input: dict[str, Any]) -> dict[str, int]:
+    score_rows = source_input.get("score_rows")
+    if not isinstance(score_rows, list) or not score_rows:
+        raise FusedSoftmaxTableRouteMatrixGateError("source score_rows missing")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)